### PR TITLE
feat: added tracing of JWT claims verification failure

### PR DESCRIPTION
**Pull request by Rakowskiii**
# Description

Log errors in verification of JWT tokens to ensure we are able to quickly debug issues.

## Due Diligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/handlers/mod.rs [hunk 1]
```diff
@@ -29,7 +30,12 @@ where
 {
     return if let Some(auth_header) = headers.get(axum::http::header::AUTHORIZATION) {
         let header_str = auth_header.to_str()?;
-        let client_id = Jwt(header_str.to_string()).decode(&HashSet::from([aud.to_string()]))?;
+        let client_id = Jwt(header_str.to_string())
+            .decode(&HashSet::from([aud.to_string()]))
+            .map_err(|e| {
+                info!("Invalid claims: {:?}", e);
```
Comment (HarryET) [inline src/handlers/mod.rs:36]: Can we get the client id here too so we can search for these logs?

**Review by HarryET — CHANGES_REQUESTED** [review]

File: src/handlers/mod.rs (hunk 1 above)
Comment (Rakowskiii) [reply src/handlers/mod.rs:36]: In here no, as this error may also be due to JWT being invalid -> not carrying client_id. We can log client_id in check closure, but then we'll need to remember to add this everywhere we provide it.

File: src/handlers/delete_client.rs [hunk 2]
```diff
@@ -27,6 +27,10 @@ pub async fn handler(
     dbg!(&state.config.public_url);
     if !authenticate_client(headers, &state.config.public_url, |client_id| {
         if let Some(client_id) = client_id {
+            info!(
```
Comment (Rakowskiii) [inline src/handlers/delete_client.rs:30]: @HarryET this is what we can do regarding client_id. Would this be sufficient?

File: src/handlers/mod.rs (hunk 1 above)
Comment (HarryET) [reply src/handlers/mod.rs:36]: Ah ok, I guess log lines should be close together so we can figure it out from that

File: src/handlers/delete_client.rs (hunk 2 above)
Comment (HarryET) [reply src/handlers/delete_client.rs:30]: Yeah, thats good. I think there is a way with log? maybe a `?` syntax not sure but that can always be done later

**Review by HarryET — APPROVED** [review]

**Pull request merged**